### DISCUSSION TITLE
Follow up to: Is there a way to easily clear all jotai atoms?

### DISCUSSION THREAD

**Discussion by pawlarius** [pmndrs/jotai#1249] [Q&A] [answered]
So someone has asked this before but I'm not sure if I understand the solution provided by the author.
My use case is I want to reset the all the jotai atoms, I tried using the solution by adding key to the Provider from [this issue](https://github.com/pmndrs/jotai/issues/404), but it's not working as expected.

I have made a sandbox here: https://codesandbox.io/s/jolly-aryabhata-ccmwm3?file=/pages/index.js
In that sandbox there are: IndexPage, ParentComponent & ChildComponent.

The use case is:
1. Click a button named: "Change desc from ChildComponent". This should change the atom `myAtom` value in the `desc` key.
2. Click a button named: "Change NAME from IndexPage", this will change the atom `myAtom` value in the `name` key that is used by the Provider as `key` and `scope`.

I assume step no 2 should reset the Provider since I changed the atom value name that is used by the Provider as both `scope` and `key`.

I'm not sure if this is a bug or my expectation is mistaken, please help enlighten me, thank you 🙇

**Comment by dai-shi** [comment, marked as answer]
(I'm not sure if `key={}` can be a solution to your case, but let's see how it works.)

- you don't need `scope`, it will be only useful if you pass the same scope to useAtom.
- the useAtom in IndexPage is outside Provider, so it may not work as you expect.

Please see this example: https://codesandbox.io/s/nice-taussig-iwd56r?file=/src/App.js

**Reply by pawlarius** [reply]
Hi, thanks for sparing time to answer this question.
In your example seems like the parent component is not relying on atom but is using different state (from React useState).

But in my case, we want to "refresh" all the other atom state if the selected key value in Provider changes.
The selected key value is only available from the children component.

So based from your example are you suggesting this isn't something that atom can support and I should use other state beside atom to "refresh" all the atom values?

**Reply by dai-shi** [reply]
React is based on declarative model, you know, If you change the state of [A, B, C] based on A change, it's mathematically not stable.

So, I guess what you want it to remount Provider. What you could do instead is to remember all atoms on your end, and reset them, or to use some kind of versionAtom to refresh all atoms making them depend on the versionAtom. (See: atomWithRefresh in advanced recipes in docs)

If we forget about the Provider, atom value is just a global variable. So, conceptually, there's no way to delete all.
```js
let atomAValue = ...
let atomBValue = ...
let atomCValue = ...
// How can you reset these three variables.
```

---
In case you or someone is interested, here's how we can use Provider scope:
https://codesandbox.io/s/magical-oskar-0jqyvw?file=/src/App.js

**Reply by pawlarius** [reply]
I see, thanks for clarifying these!

**Comment by AjaxSolutions** [comment]
I use the onMount function because there are some atoms which I don't want to clear.
```
derivedAtom.onMount = (setAtom) => {
    setAtom({ type: ATOM_ACTION.initialize });
    return () => {
        setAtom({ type: ATOM_ACTION.reset });
    };
};
```
See the [unMount](https://jotai.org/docs/api/core#on-mount) doc.

**Reply by pawlarius** [reply]
Hi, thanks for joining the discussion!
I checked the docs and seems like the unMount is called "when it’s no longer used".

Can you help clarify what does it meant by "no longer used"?

**Comment by jobinabraham** [comment]
Consder the following use case with react native app and wix [RNN](https://github.com/wix/react-native-navigation)  :

1. User opens app
2. User logs into the app 
3. A new root is set with  5 bottomtabs 
4. Each component registered in the bottomtab has its own tree.(No common ancestor) 
5. All the 5 screens use some remote data that can be asynchronously loaded 
6. All 5 screens mount parallely -> exporting a loadable atom + atomWithDefault config seems a good way to prevent multiple api calls to fetch the same data across screens 
7. On logout the data needs to be cleared **(how?)**
8. User logs into the app as another user, reinitialise the atom config **(how?)**

**Reply by dai-shi** [reply]
How would you do it with useState+useContext?
